test/libsolidity/util/ContractABIUtils.cpp

```diff
@@ -281,26 +281,18 @@ void ContractABIUtils::overwriteParameters(
 )
 {
 	using namespace placeholders;
-	boost::for_each(
-		_sourceParameters,
-		_targetParameters,
-		std::bind<void>(
-			[&](Parameter _a, Parameter& _b) -> void
-			{
+	for (std::tuple<Parameter, Parameter&> params : ranges::views::zip(_sourceParameters, _targetParameters)) {
```

Comment:
```suggestion
	for (auto params : ranges::views::zip(_sourceParameters, _targetParameters)) 
	{
```
I would rather have the explicit type being declared, but it seems that creates a problem with the OSX compiler (maybe a version problem? needs further investigation). So, at least for now, I think `auto` should fix it.  
We should confirm with other team members if that it's ok, but it would be a temporary quick fix.
Other than that, just a minor style correction.

Reply:
I don't think you can really store a reference like that in a tuple. It's not assignable. It's rather weird that GCC even allows that.

This should really look like this:
```
for (auto&& [source, target]: ranges::views::zip(_sourceParameters, _targetParameters))
```
And in this case `auto` is mandatory.

Reply:
Thank you guys. 
Sorry cameel, what's the version that shouldn't even compile? I had the three of them compile just fine with GCC 11.2.0 x86_64-linux-gnu. 
This one
`for  for (std::tuple<Parameter, Parameter&> params : ranges::views::zip ...`
or this one 
`for  for (auto  params : ranges::views::zip ...`

Reply:
I mean anything with `std::tuple<Parameter, Parameter&>` in it. Well, I can't say for sure that it must not compile because there could be some template magic that makes it work somehow, but I'm just going by the general notion that you normally can't store references in a container. You'd have to use a workaround like `std::reference_wrapper`. And anyway, `ranges` probably returns one of it's own wrappers here and these types are usually long and ugly - it's one situation where we actually do prefer `auto` to hide the nitty-gritty details.

In any case, please try to use the version I posted above.